src/Aspire.Dashboard/Otlp/Model/OtlpHelpers.cs

```diff
@@ -13,8 +14,27 @@
 
 namespace Aspire.Dashboard.Otlp.Model;
 
-public static class OtlpHelpers
+public static partial class OtlpHelpers
 {
+    private static readonly string s_longTimePatternWithMilliseconds = GetLongTimePatternWithMilliseconds();
+
+    static string GetLongTimePatternWithMilliseconds()
+    {
+        // From https://learn.microsoft.com/dotnet/standard/base-types/how-to-display-milliseconds-in-date-and-time-values
+
+        // Gets the long time pattern, which is something like "h:mm:ss tt" (en-US), "H:mm:ss" (ja-JP), "HH:mm:ss" (fr-FR).
+        var longTimePattern = DateTimeFormatInfo.CurrentInfo.LongTimePattern;
+
+        // Create a format similar to .fff but based on the current culture.
+        var millisecondFormat = $"{NumberFormatInfo.CurrentInfo.NumberDecimalSeparator}fff";
+
+        // Append millisecond pattern to current culture's long time pattern.
+        return MatchSecondsInTimeFormatPattern().Replace(longTimePattern, $"$1{millisecondFormat}");
```

Comment:
If the replace fails, would it be better to use the default LongTimePattern, or fall back to a hardcoded value?, e.g. `"h:mm:ss.fff tt"`

Reply:
I do not expect the replace to fail. I cannot find a way to remove seconds from the long time format. The addition of seconds is the primary difference between long and short time formats. If the user has somehow requested to never see seconds in their timestamps, then that's probably what they want here.